src/Trinity.C/src/Storage/MTHash/MTHash.CellInfo.cpp

```diff
@@ -75,18 +111,21 @@ namespace Storage
                 entryIndex = entry_index;
                 /////////////////////////
 
-                CellEntry entry = { updated_cell_offset, size };
+                CellEntry entry ={ updated_cell_offset, size };
                 std::atomic<int64_t> * CellEntryAtomicPtr = (std::atomic<int64_t>*) CellEntries;
-                (CellEntryAtomicPtr + entry_index)->store(entry.location);
+
+                if (eResult == TrinityErrorCode::E_SUCCESS)
+                {
+                    (CellEntryAtomicPtr + entry_index)->store(entry.location);
+                    if (CellTypeEnabled)
+                        MTEntries[entry_index].CellType = type;
+                }
 
                 /// add_memory_entry_flag epilogue
                 LEAVE_ALLOCMEM_CELLENTRY_UPDATE_CRITICAL_SECTION();
                 /// add_memory_entry_flag epilogue
 
-                if (CellTypeEnabled)
-                    MTEntries[entry_index].CellType = type;
-
-                return TrinityErrorCode::E_SUCCESS;
+                return eResult;
```

Comment:
>eResult; [](start = 23, length = 8)

Because the function does not return E_SUCCESS any more, can we ensure that all the caller functions have been updated accordingly to handle the TrinityErrorCode correctly? It would be better to have test on it.